async-gateway/storage.go

```diff
@@ -80,6 +82,25 @@ func (s *s3) Download(key string) ([]byte, error) {
 	return buff.Bytes(), nil
 }
 
+// List lists a set of files from a given S3 path
+func (s *s3) List(key string) ([]string, error) {
```

Comment:
Nit: it would be good to update the comment and/or the name of the function to indicate that this only works if all sub-paths are only one level deep. Also it might be worth ensuring that the input prefix ends in a slash (assuming it's not empty string)

Reply:
Done. Also, for ensuring the key ends with a "/", I added the logic inside the function for that without creating a utility function like `EnsureSuffix`. If we end up needing that in multiple places, that's when we could add the utility instead.